hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -203,13 +205,17 @@ private void balance() {
           } catch (InterruptedException e) {
             LOG.info("Container Balancer was interrupted while waiting for" +
                 " next iteration.");
-            stop();
+            if (!isBalancerRunning()) {
+              stop();
+            }
             return;
           }
         }
       }
     }
-    stop();
+    if (!isBalancerRunning()) {
```

Comment:
seems we should remove `!` here , or there may be a dead lock

Reply:
Yes, I meant to check `isBalancerRunning` without the `!` to avoid a deadlock. Thanks for pointing this mistake out!